docs/contributing/sources-and-providers.md

```diff
@@ -21,6 +21,7 @@ All sources live in package `source`.
 
 * `ServiceSource`: collects all Services that have an external IP and returns them as Endpoint objects. The desired DNS name corresponds to an annotation set on the Service or is compiled from the Service attributes via the FQDN Go template string.
 * `IngressSource`: collects all Ingresses that have an external IP and returns them as Endpoint objects. The desired DNS name corresponds to the host rules defined in the Ingress object.
+* `GatewaySource`: collects all Istio Gateways and returns them as Endpoint objects. The desired DNS name corresponds to the hosts listed within the servers spec of each Gateway object.
```

Comment:
Should be names IstioGatewaySource, GatewaySource is to generic

Reply:
Nice suggestion and good point. I'll make the change across the board.